html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,82 @@
 <?php
 $limit = $_GET['target'];
 
-$dsn = 'mysql:dbname=test;host=mysql';
-$dbuser = 'test';
-$dbpassword = 'test';
+if(strpos($limit,'.0')){
+    http_response_code(400);
+    exit();
+}
+
+if($limit <= 0 or $limit != (integer)$limit or $limit != is_numeric($limit)){
+    http_response_code(400);
+    exit();
+}else{
+    $dsn = 'mysql:dbname=test;host=mysql';
+    $dbuser = 'test';
+    $dbpassword = 'test';
+    try{
+        $db = new PDO($dsn,$dbuser,$dbpassword);
+    }catch (PDOException $e){
+        echo 'DB接続エラー:'.$e->getMessage();
+        http_response_code(404);
+        exit();
+    }
+    $records = $db->query('SELECT value FROM prechallenge3');
+    $record = $records->fetchAll(PDO::FETCH_ASSOC);
+
+    //一旦valuカラムから全部の数字を取り出して単次元配列にする処理
+    $array_amount = count($record);
+        for($i = 0; $i < $array_amount; $i++){
+            $record_array[] = (int)$record[$i]['value'];
+        }
```

Comment:
for全体のインデントが一つずれています。
```suggestion
    $array_amount = count($record);
    for($i = 0; $i < $array_amount; $i++){
        $record_array[] = (int)$record[$i]['value'];
    }
```